.circleci/config.yml

```diff
@@ -30,6 +30,9 @@ jobs:
               done
               git checkout yarn.lock
             fi
+      - run:
+          name: Build typechain types
+          command: TS_NODE_TRANSPILE_ONLY=1 hardhat typechain
```

Comment:
this seems redundant since both `yarn` runs this via the `prepare` lifecycle hook (`yarn prepare` is run before all yarn installs, and our `prepare` hook calls `yarn build` which in turn calls `hardhat compile` which finally in turn internally calls `hardhat typechain`). That said, I noticed you set the default for most scripts to use `TS_NODE_TRANSPILE_ONLY`. I don't think this is a good idea since all type safety is permanently lost and we'll no longer hear about issues with types in almost all cases. I previously added `yarn clean` to circuvent the type issue we sometimes see, and so if we absolutely need to circuvent type issues we should use that before calling whatever script. You usually only need to run `yarn clean` once to circumvent the type issues and then in subsequent calls you can reliably use the type-safe vairants.

Reply:
Not sure why but I very infrequently run into it. Wonder if others are doing something differently

Reply:
if nothing else, I would add this to the top of the file

```yaml
commands:
  build-typechain-types:
    description: 'Build typechain types'
    steps:
      - run:
          name: Run typechain with TS in transpile-only mode
          command: |
            TS_NODE_TRANSPILE_ONLY=1 hardhat typechain # todo figure out how to get this to work without `TS_NODE_TRANSPILE_ONLY`
```

Reply:
then use it like

```yaml
  run-tests:
    <<: *defaults
    steps:
      - build-typechain-types
```